src/js/utils/general.js

```diff
@@ -97,7 +97,7 @@ export function uuid() {
  * @private
  */
 function _makeTooltipInstance(attachToOptions, step) {
-  let tetherOptions = {
+  let tetherOptions = Object.assign({}, {
```

Comment:
The problem with this is `Object.assign` only works one level deep, so anything where an object is used for the property could have issues merging different keys inside that option. For example:

```js
new Tether({
  element: yellowBox,
  target: greenBox,
  attachment: 'top left',
  targetAttachment: 'bottom left',
  optimizations: {
    moveElement: false
  }
});
```

I think for anything that is a hash, like `optimizations` or `classes` we want to manually add those in, to ensure we can merge the nested options and overwrite where necessary.

Does that make sense? We had something similar for modifiers in the old `popperOptions`, if you take a look at the history.

Reply:
Oopsy, I didn’t know about this, maybe it makes sense to use the lodash function just for this case!

-> https://lodash.com/docs#merge

Reply:
@lex111 we don't want to add lodash here.

I would recommend something like:

```js
  tetherOptions.optimizations = {
    ...tetherOptions.optimizations,
    ...step.options.tetherOptions.optimizations,
  };
```

This is just an example, and I have not tried it.

Reply:
Got you, but we don’t need to import the whole library, if I’m not mistaken, its functions can be used separately (as packages), so this will not increase the bundle size.

Reply:
They can be used separately, but we do not want to add them

Reply:
@rwwagner90 sorry for the long wait, I did it, please take a look.